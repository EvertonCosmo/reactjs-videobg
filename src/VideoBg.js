import React from "react";
import classNames from "./VideoBg.module.css";
import Source from "./Source";
import PropTypes from "prop-types";

const VideoBg = ({
  wrapperClass,
  videoClass,
  loop = true,
  muted = true,
  autoPlay = true,
  poster,
  children
}) => (
  <div className={`${wrapperClass} ${classNames.wrapper}`}>
    <video
      className={`${videoClass} ${classNames.video}`}
      loop={loop}
      muted={muted}
      poster={poster}
      autoPlay={autoPlay}
    >
      {children}
    </video>
  </div>
);

VideoBg.propTypes = {
  wrapperClass: PropTypes.string,
  videoClass: PropTypes.string,
  loop: PropTypes.string,
  muted: PropTypes.string,
  poster: PropTypes.string,
  autoPlay: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.shape({
      type: Source
    }),
    PropTypes.arrayOf(
      PropTypes.shape({
        type: Source
      })
    )
  ])
};

VideoBg.Source = Source;

export default VideoBg;
