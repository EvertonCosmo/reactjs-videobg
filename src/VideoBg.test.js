import React from "react";
import VideoBg from "./VideoBg";
import renderer from "react-test-renderer";

it("renders properly", () => {
  const tree = renderer.create(<VideoBg />).toJSON();
  expect(tree).toMatchSnapshot();
});
