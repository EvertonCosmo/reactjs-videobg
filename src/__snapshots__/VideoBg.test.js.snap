// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="undefined undefined"
>
  <video
    autoPlay={true}
    className="undefined undefined"
    loop={true}
    muted={true}
  />
</div>
`;
